src/execution_plan/ops/op_create.c

```diff
@@ -44,40 +50,37 @@ static void _PendingPropertiesFree(PendingProperties *props) {
 	rm_free(props);
 }
 
-OpBase *NewCreateOp(ResultSetStatistics *stats, NodeCreateCtx *nodes, EdgeCreateCtx *edges) {
-	OpCreate *op_create = calloc(1, sizeof(OpCreate));
-	op_create->records = NULL;
-	op_create->nodes_to_create = nodes;
-	op_create->edges_to_create = edges;
-	op_create->gc = QueryCtx_GetGraphCtx();
-	op_create->created_nodes = array_new(Node *, 0);
-	op_create->created_edges = array_new(Edge *, 0);
-	op_create->node_properties = array_new(PendingProperties *, 0);
-	op_create->edge_properties = array_new(PendingProperties *, 0);
-	op_create->stats = stats;
+OpBase *NewCreateOp(const ExecutionPlan *plan, ResultSetStatistics *stats, NodeCreateCtx *nodes,
+					EdgeCreateCtx *edges) {
+	OpCreate *op = calloc(1, sizeof(OpCreate));
+	op->records = NULL;
+	op->nodes_to_create = nodes;
+	op->edges_to_create = edges;
+	op->gc = QueryCtx_GetGraphCtx();
+	op->created_nodes = array_new(Node *, 0);
+	op->created_edges = array_new(Edge *, 0);
+	op->node_properties = array_new(PendingProperties *, 0);
+	op->edge_properties = array_new(PendingProperties *, 0);
+	op->stats = stats;
 
 	// Set our Op operations
-	OpBase_Init(&op_create->op);
-	op_create->op.name = "Create";
-	op_create->op.type = OPType_CREATE;
-	op_create->op.consume = OpCreateConsume;
-	op_create->op.init = OpCreateInit;
-	op_create->op.reset = OpCreateReset;
-	op_create->op.free = OpCreateFree;
+	OpBase_Init((OpBase *)op, OPType_CREATE, "Create", CreateInit, CreateConsume,
+				CreateReset, NULL, CreateFree, plan);
 
 	uint node_blueprint_count = array_len(nodes);
 	uint edge_blueprint_count = array_len(edges);
+
 	// Construct the array of IDs this operation modifies
-	op_create->op.modifies = array_new(uint, node_blueprint_count + edge_blueprint_count);
 	for(uint i = 0; i < node_blueprint_count; i ++) {
-		op_create->op.modifies = array_append(op_create->op.modifies, nodes[i].node_idx);
+		nodes[i].node_idx = OpBase_Modifies((OpBase *)op, nodes[i].node->alias);
 	}
 	for(uint i = 0; i < edge_blueprint_count; i ++) {
-		// TODO should this also add the src and dest IDs?
-		op_create->op.modifies = array_append(op_create->op.modifies, edges[i].edge_idx);
+		edges[i].edge_idx = OpBase_Modifies((OpBase *)op, edges[i].edge->alias);
+		edges[i].src_idx = OpBase_Modifies((OpBase *)op, edges[i].edge->src->alias);
+		edges[i].dest_idx = OpBase_Modifies((OpBase *)op, edges[i].edge->dest->alias);
```

Comment:
`MATCH (a),(b) CREATE (a)-[e]->(b)`
Please make sure `op_create` doesn't consider itself as a modifier of either `a` or `b`.